modules/docs/docgen/spec/enhancedComponentParser.spec.ts

```diff
@@ -0,0 +1,720 @@
+
+      expect(symbols).toHaveProperty('0.name', 'MyComponent');
+      expect(symbols).toHaveProperty('0.type.kind', 'enhancedComponent');
+      expect(symbols).toHaveProperty('0.type.displayName', 'My.Component');
+      expect(symbols).toHaveProperty('0.type.elemPropsHook', 'useMyComponent');
+      expect(symbols).toHaveProperty('0.type.baseElement.kind', 'external');
+      expect(symbols).toHaveProperty('0.type.baseElement.name', 'div');
+      expect(symbols).toHaveProperty('0.type.props.0.kind', 'property');
+      expect(symbols).toHaveProperty('0.type.props.0.name', 'size');
+      expect(symbols).toHaveProperty('0.type.props.0.type.kind', 'primitive');
+      expect(symbols).toHaveProperty('0.type.props.0.type.value', 'string');
+      expect(symbols).toHaveProperty('0.type.props.0.defaultValue.kind', 'string');
+      expect(symbols).toHaveProperty('0.type.props.0.defaultValue.value', 'medium');
+      expect(symbols).toHaveProperty('0.type.model', 'MyModel');
+      expect(symbols).toHaveProperty('0.type.subComponents.0.name', 'Item');
+      expect(symbols).toHaveProperty('0.type.subComponents.0.symbol', 'MyComponentItem');
+    });
+
+    it('should handle "elemPropsHook" that is aliased', () => {
+      const program = createProgramFromSource(
+        'test.tsx',
+        `
+        export const MyComponent = createContainer('div')({
+          displayName: 'My.Component',
+          modelHook: useMyModel,
+          elemPropsHook: useMyComponent,
+        })<MyComponentProps>((elemProps, Element) => {
+          return <Element size="medium" {...elemProps} />
+        });
+
+        export const useMyComponent = useMyOtherComponent
+
+        export interface MyComponentProps {
+          size: string
+        }
+      `
+      );
+
+      const symbols = parse(program, 'test.tsx', [enhancedComponentParser]);
+
+      expect(symbols).toHaveProperty('0.name', 'MyComponent');
+      expect(symbols).toHaveProperty('0.type.kind', 'enhancedComponent');
+      expect(symbols).toHaveProperty('0.type.displayName', 'My.Component');
+      expect(symbols).toHaveProperty('0.type.elemPropsHook', 'useMyOtherComponent');
+      expect(symbols).toHaveProperty('0.type.baseElement.kind', 'external');
+      expect(symbols).toHaveProperty('0.type.baseElement.name', 'div');
+      expect(symbols).toHaveProperty('0.type.props.0.kind', 'property');
+      expect(symbols).toHaveProperty('0.type.props.0.name', 'size');
+      expect(symbols).toHaveProperty('0.type.props.0.type.kind', 'primitive');
+      expect(symbols).toHaveProperty('0.type.props.0.type.value', 'string');
+      expect(symbols).toHaveProperty('0.type.props.0.defaultValue.kind', 'string');
+      expect(symbols).toHaveProperty('0.type.props.0.defaultValue.value', 'medium');
+      expect(symbols).toHaveProperty('0.type.model', 'MyModel');
+    });
+
+
+    it('should handle "elemPropsHook" that is aliased', () => {
```

Comment:
This has the same description as the previous test. A couple things are different but I noticed this one has a subcomponent and no elemsPropHook.